src/databricks/labs/ucx/assessment/azure.py

```diff
@@ -129,33 +126,40 @@ def _safe_get_cluster_policy(self, policy_id: str) -> Policy | None:
     def _get_azure_spn_from_config(self, config: dict) -> set[AzureServicePrincipalInfo]:
         """Detect azure service principals from a dictionary of spark configs
         This checks for the existence of an application id by matching fs.azure.account.oauth2.client.id
-        For each application id identified, retrieve the relevant storage account, and corresponding tenant id
+        For each application id identified, retrieve the relevant storage account, client secret and tenant id
         """
         set_service_principals = set[AzureServicePrincipalInfo]()
-        spn_application_id, secret_scope, secret_key, tenant_id, storage_account = None, "", "", "", ""
         matching_spn_app_id_keys = [key for key in config.keys() if "fs.azure.account.oauth2.client.id" in key]
         for spn_app_id_key in matching_spn_app_id_keys:
             # retrieve application id of spn
-            spn_application_id, secret_scope, secret_key = self._get_key_from_config(spn_app_id_key, config)
+            spn_application_id = self._get_value_from_config_key(config, spn_app_id_key)
             if spn_application_id is None:
                 continue
 
+            storage_account, secret_scope, secret_key, tenant_id = None, None, None, None
+            tenant_key = "fs.azure.account.oauth2.client.endpoint"
+            client_secret_key = "fs.azure.account.oauth2.client.secret"
+
             # retrieve storage account configured with this spn
             storage_account_matched = re.search(STORAGE_ACCOUNT_EXTRACT_PATTERN, spn_app_id_key)
             if storage_account_matched:
                 storage_account = storage_account_matched.group(1).strip(".")
-                tenant_key = "fs.azure.account.oauth2.client.endpoint." + storage_account
+                # adjust the key to lookup for tenant id & secret id
+                tenant_key = tenant_key + f".{storage_account}"
```

Comment:
```suggestion
                tenant_key =
f"{tenant_key}.{storage_account}"
```
More readable this way